### ISSUE TITLE
Measure for a function of two arguments

### ISSUE THREAD

**Issue by anton-trunov** [mattam82/Coq-Equations#42]
I'm trying to define the `interleave` function as follows:

```
From Coq Require Import Arith List.
From Equations Require Import Equations.

Equations interleave {A} (l l' : list A) : list A :=
interleave l l' by rec (length l + length l') lt :=
interleave nil l' := l';
interleave (cons a l) l' := a :: interleave l' l.
Next Obligation.
```

But for some reason `Equations` unfolds the `add` and `length` functions, so I see the following obligation:
```
  (fix add (n m : nat) {struct n} : nat :=
     match n with
     | 0 => m
     | S p => S (add p m)
     end)
    ((fix length (l0 : list A1) : nat :=
        match l0 with
        | [] => 0
        | _ :: l'0 => S (length l'0)
        end) l')
    ((fix length (l0 : list A1) : nat :=
        match l0 with
        | [] => 0
        | _ :: l'0 => S (length l'0)
        end) l) <
  S
    ((fix add (n m : nat) {struct n} : nat :=
        match n with
        | 0 => m
        | S p => S (add p m)
        end)
       ((fix length (l0 : list A1) : nat :=
           match l0 with
           | [] => 0
           | _ :: l'0 => S (length l'0)
           end) l)
       ((fix length (l0 : list A1) : nat :=
           match l0 with
           | [] => 0
           | _ :: l'0 => S (length l'0)
           end) l'))
```
instead of
```
length l' + length l < length (a :: l) + length l'
```

How do I prevent this?

**Comment by mattam82** [comment]
Fix is merged in 8.8, needs to be backported in 8.6/8.7

**Comment by cmangin** [comment]
Fixed in master and 8.7 too. Thanks for the report!

**Comment by anton-trunov** [comment]
Thanks for fixing it!
